geode-core/src/main/java/org/apache/geode/distributed/internal/membership/gms/messenger/JGroupsMessenger.java

```diff
@@ -537,9 +537,7 @@ private void establishLocalAddress() {
         || !config.getStartLocator().isEmpty();
 
     // establish the DistributedSystem's address
-    String hostname =
```

Comment:
You'll need to run some rolling upgrade tests with clients having non-durable CQs and servers having enable-network-partition-detection=true to test this change.